_posts/2020-04-07-trying-out-julia.md

```diff
@@ -0,0 +1,227 @@
+
+Plotting is actually the "entry point" for this blog. I'll try to do same kind of stuff I did with MatLab with Julia. So basically this means scripting some plotting. So I won't delve into the finer points of e.g. building server side software.
+
+So if you want deep knowledge and insight into Julia as a language this is not the blog post for you, but if you just want to know how to display some data and setup your own Julia environment please do continue.
+
+## Okay how to install this thing
+
+Well let's google it. The page [julialang.org/](https://julialang.org/) is the home page and it has a big download button. Straight away the installation is a bit bare bones since it's just a tar-package (mind you I'm using Linux based system) and not some fancy installer or CLI-tool. After I've added the extracted bin folder to my PATH it's time to take the first sip with the repl.
+
+![Julia REPL](/img/trying-out-julia/julia-repl.png)
+
+Great we have a fancy calculator which seems to be even calculating correctly! After some 20-20 hindsight I can say there is some important information already displayed in the welcome ASCII-art, mainly the ]-mark. This is the way to access Julia package manager for adding dependencies. Also with hindsight using `activate <env_name>` with the package manager allows you to create python like virtual environments to contain your packages within your project.
+
+So good so far but I'm one of those guys who likes IDEs especially when learning new languages. The Julia homepage has a nice list of supported and familiar editors and IDEs. But why use something you already know? Let's go all in with new stuff and select Atom editor with JUNO plug-in. Ok yes part of the reason selecting Juno was that it looked cool in screenshots at [junolab.org](https://junolab.org). The installation was quite straight forward as Atom had a .deb-package and after installing that I just needed to add the uber-juno plug-in with atom's built in tools. Yes now I had all the tools in place, time to start coding.
+
+## What to do
+
+As said I have some background with MatLab and wanted to do some visualizations as they seem to be the strong point of Julia. So what data to visualize, something current...surprise surprise let's use COVID-19 data. Basic status data can be acquired from [api.covid19api.com](https://api.covid19api.com/) using REST calls. So basically to get day to day confirmed cases from Finland issue GET call to [api.covid19api.com/country/finland/status/confirmed](https://api.covid19api.com/country/finland/status/confirmed) and you get an array of the following json objects
+
+```json
+{
+    "Country": "Finland",
+    "Province": "",
+    "Lat": 61.9241,
+    "Lon": 25.7482,
+    "Date": "2020-03-31T00:00:00Z",
+    "Cases": 1418,
+    "Status": "confirmed"
+},
+```
+
+After some document browsing I know I need to add some packages to my Julia environment 
```

Comment:
After some intensive documentation browsing/skimmin/reading, I found out that I had to add some additional packages to my Julia environment.